internal/plugin/connectors/http/generic/connector.go

```diff
@@ -39,11 +40,23 @@ func (c *Connector) Connect(
 	}
 
 	// Add configured params to request
-	params, err := renderTemplates(c.config.QueryParams, credentialsByID)
+	queryParams, err := renderTemplates(c.config.QueryParams, credentialsByID)
 	if err != nil {
 		return fmt.Errorf("failed to render query params: %s", err)
 	}
-	r.URL.RawQuery = appendQueryParams(*r.URL, params)
+	r.URL.RawQuery = appendQueryParams(*r.URL, queryParams)
+
+	// Add OAuth1 Authorization Header to the request
+	oauth1params, err := renderTemplates(c.config.OAuth1Secrets, credentialsByID)
+	if err != nil {
+		return fmt.Errorf("failed to render OAuth1 secrets: %s", err)
+	} else {
```

Comment:
If you return from the function, the `else` is implied already
```suggestion
	if err != nil {
		return fmt.Errorf("failed to render OAuth1 secrets: %s", err)
	}
	
	 <other code>
```